### PR TITLE
[WIP] Add kpod-run/kpod-create commands

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@mheon This should probably be combined with the other kpod-create PR, but I thought I would start the process of writing man pages and bash completions.  Basically grabbed the content from docker run/create man pages and converted them to kpod.  We need to look through these man pages to figure out options that we do not plan to implement.

**Comment by rhatdan** [comment]
@baude @umohnani8 PTAL

**Comment by rhatdan** [comment]
@runcom @mrunalp PTAL, please indicate commands that we probably don't want to implement.

**Comment by mheon** [comment]
I don't like --security-opt. I feel like it would be better to replace it with more specific seccomp, SELinux, Apparmor flags, instead of lumping everything under one and parsing it ourselves to figure out what's going on.

**Comment by rhatdan** [comment]
@mheon The goal is to stay in step with the Docker CLI. If someone sees on google how to disable SELinux is to execute

docker run --security-opt label=disable ...

Then we want them to only have to do 
kpod run --security-opt label=disable ...

If you want to later expand or add some other commands to do the same thing I am fine with it. But I want to match docker CLI.  I would like to be able to pass most of the docker cli test suite.

**Comment by mheon** [comment]
I think we can also prune `--restart` and `--link` (and maybe similar commands related to container communication). We don't have a daemon, so taking care of container restarts is a little problematic - should leave that to systemd, I think. And container IPC can be handled by pods, which will probably end up with their own commands and/or flags.

I'll do a more thorough review tomorrow.

**Comment by rhatdan** [comment]
Thanks I was thinking the same thing.

**Comment by rhatdan** [comment]
This patch pulls in https://github.com/kubernetes-incubator/cri-o/pull/977

File: cmd/kpod/create.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+type createResourceConfig struct {
+	blkioWeight       int64
+	blkioDevice       []string
+	cpuShares         int64
+	cpuCount          int64
+	cpuPeriod         int64
+	cpusetCpus        string
+	cpusetNames       string
+	cpuFile           string
+	cpuMems           string
```
Comment (baude) [inline cmd/kpod/create.go]: What are the correlating command line options for this one and the two above it?

File: cmd/kpod/create.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	deviceWriteBps    []string
+	deviceWriteIops   []string
+	memory            string
+	memoryReservation string
+	memorySwap        string
+	memorySwapiness   string
+	kernelMemory      string
+	oomScoreAdj       string
+	pidsLimit         string
+	shmSize           string
```
Comment (baude) [inline cmd/kpod/create.go]: This is duplicated below in the createConfig struct.  Is that done with intent?

File: cmd/kpod/create.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	ulimit            []string
+}
+
+type createConfig struct {
+	additionalGroups []int64
```
Comment (baude) [inline cmd/kpod/create.go]: What is the correlating command line switch for this?  There is a groupAdd below as well.

File: cmd/kpod/create.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+	image            string
+	interactive      bool
+	ip6Address       string
+	ipAddress        string
+	labels           map[string]string
+	linkLocalIP      []string
+	logDriver        string
+	logDriverOpt     []string
+	macAddress       string
+	mounts           []*pb.Mount
```
Comment (baude) [inline cmd/kpod/create.go]: What is the correlating command line switch for mounts?

File: cmd/kpod/create.go [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+	name             string
+	network          string
+	networkAlias     []string
+	nsIPC            string
+	nsNet            string
+	nsPID            string
+	nsUser           string
```
Comment (baude) [inline cmd/kpod/create.go]: What is the correlating command line switch for nsUser? There is a userns as well.

File: cmd/kpod/create.go [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+	pod              string
+	ports            []*pb.PortMapping
```
Comment (baude) [inline cmd/kpod/create.go]: What is the correlating command line switch for ports?

File: cmd/kpod/create.go [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+	privileged       bool
+	publish          []string
+	publishAll       bool
+	readOnlyRootfs   bool
+	resources        createResourceConfig
+	rm               bool
+	securityOpts     []string
+	shmSize          string
+	sigProxy         bool
+	stdin            bool
```
Comment (baude) [inline cmd/kpod/create.go]: What is the correlating command line switch for stdin?

File: cmd/kpod/create.go (hunk 1 above)
Comment (rhatdan) [reply cmd/kpod/create.go]: Removing, leftovers.

File: cmd/kpod/create.go (hunk 2 above)
Comment (rhatdan) [reply cmd/kpod/create.go]: Removed below.

File: cmd/kpod/create.go (hunk 3 above)
Comment (rhatdan) [reply cmd/kpod/create.go]: Nice catch, removing.

File: cmd/kpod/create.go (hunk 4 above)
Comment (rhatdan) [reply cmd/kpod/create.go]: Volumes, I guess, so removing.

File: cmd/kpod/create.go (hunk 5 above)
Comment (rhatdan) [reply cmd/kpod/create.go]: Lets stay with nsUser for consistency

File: cmd/kpod/create.go (hunk 6 above)
Comment (rhatdan) [reply cmd/kpod/create.go]: Lets leave it as publish I guess for right now.

File: cmd/kpod/create.go (hunk 7 above)
Comment (rhatdan) [reply cmd/kpod/create.go]: Removing

File: cmd/kpod/create.go (hunk 6 above)
Comment (mheon) [reply cmd/kpod/create.go]: I don't know if we can use pb.PortMapping here - Kube only allows single ports, Docker allows ranges. So the Kube PortMapping struct probably won't work here.

Comment (rhatdan) [reply cmd/kpod/create.go]: I removed it anyways.

**Comment by rhatdan** [comment]
Replaced by https://github.com/kubernetes-incubator/cri-o/pull/1005

**Pull request closed without merging**